File: app/src/main/java/com/kidozh/discuzhub/results/BuyThreadResult.java
package com.kidozh.discuzhub.results;

import androidx.annotation.NonNull;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

public class BuyThreadResult extends BaseResult {
    @JsonProperty("Variables")
    @NonNull
    public BuyThreadVariableResult variableResults;

    public static class BuyThreadVariableResult extends VariableResults{
        @JsonProperty("authorid")
        @JsonFormat(shape = JsonFormat.Shape.STRING)
        public int authorId;
        public String author;
        @JsonFormat(shape = JsonFormat.Shape.STRING)
        public int price, balance;
        public Credit credit;

    }

    public static class Credit{
        public String title = "";
        @JsonIgnoreProperties(ignoreUnknown = true)
        public String unit = "";
    }
}

File: app/src/main/java/com/kidozh/discuzhub/results/BuyThreadResult.kt
package com.kidozh.discuzhub.results

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

class BuyThreadResult : BaseResult() {
    @JsonProperty("Variables")
    var variableResults: BuyThreadVariableResult = BuyThreadVariableResult()

    class BuyThreadVariableResult : VariableResults() {
        @JsonProperty("authorid")
        @JsonFormat(shape = JsonFormat.Shape.STRING)
        var authorId = 0
        var author: String? = null

        @JsonFormat(shape = JsonFormat.Shape.STRING)
        var price = 0

        @JsonFormat(shape = JsonFormat.Shape.STRING)
        var balance = 0
        var credit: Credit = Credit()
    }

    class Credit {
        var title = ""

        @JsonIgnoreProperties(ignoreUnknown = true)
        var unit = ""
    }
}